### PR TITLE
Dash/connector kpi

### PR DESCRIPTION
**Pull request by pawellysy**
Implemented usage of connectors and sync (extremes event) for KPI component.

### CODE REVIEW

File: ts/Dashboards/Components/KPIComponent.ts [hunk 1]
```diff
@@ -420,47 +427,55 @@ class KPIComponent extends Component {
...
+        this.setOptions();
         if (options.chartOptions && this.chart) {
             this.chart.update(options.chartOptions);
         }
 
-        this.redraw();
+        redraw && this.redraw();
     }
 
+    public getValue(): string|number|undefined {
```
Comment (karolkolodziej) [inline ts/Dashboards/Components/KPIComponent.ts:456]: The description is missing. Should it be public?

File: ts/Dashboards/Components/KPIComponent.ts [hunk 2]
```diff
@@ -420,47 +427,55 @@ class KPIComponent extends Component {
...
+
+        return this.options.value;
+    }
     /**
      * Handles updating elements via options
      *
      * @internal
      */
-    private updateElements(): void {
+
```
Comment (karolkolodziej) [inline ts/Dashboards/Components/KPIComponent.ts:472]: space

File: ts/Dashboards/Components/KPIComponent.ts [hunk 3]
```diff
@@ -476,11 +491,29 @@ class KPIComponent extends Component {
             }
 
             AST.setElementHTML(this.value, value);
-            AST.setElementHTML(this.subtitle, this.getSubtitle());
 
             this.prevValue = prevValue;
         }
+    }
+    private updateElements(): void {
```
Comment (karolkolodziej) [inline ts/Dashboards/Components/KPIComponent.ts:498]: Description

**Review by karolkolodziej — CHANGES_REQUESTED** [review]

**Review by karolkolodziej — APPROVED** [review]

**Review by sebastianbochan — APPROVED** [review]

**Comment by sebastianbochan** [comment]
Please lint ;)

**Pull request merged**